client/client-lib/src/lib.rs

```diff
@@ -554,14 +555,12 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
             let (mut keys, input) = MintClient::ecash_input(notes)?;
             tx.input(&mut keys, input);
             self.submit_tx_with_change(tx, rng).await?;
-            self.fetch_all_notes().await;
+            self.fetch_all_notes().await?;
```

Comment:
Failing if a part fails is not optimal here imo:

1. The failure could be from another tx
2. The timeout is too tight to call fetch right after submit (2s, in real scenarios epochs take longer), so this will always fail. Needs to re-try imo.

Reply:
What I'd do:

1. Only fetch the two outputs that were just generated
2. Increase timeout so that the necessary epochs can happen in that time in realistic scenarios